### PR TITLE
fx quant: add quantized Softmax workflow integration

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#75106 fx quant: add quantized Softmax workflow integration**

Summary:

In https://github.com/pytorch/pytorch/pull/75017 a quantized softmax
kernel was added. This PR adds the FX graph mode quantization workflow
integration to swap `nn.Softmax` to `nnq.Softmax`.

Test plan:

```
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops
```

Differential Revision: [D35324817](https://our.internmc.facebook.com/intern/diff/D35324817)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

File: test/quantization/fx/test_quantize_fx.py [hunk 1]
```diff
@@ -6002,6 +6003,8 @@ def forward(self, x):
                 # F.tanh is deprecated
                 x = torch.tanh(x)
                 x = x.tanh()
+                # TODO(future PR): handle F.softmax
```
Comment (jerryzh168) [inline test/quantization/fx/test_quantize_fx.py:6006]: we can wait a bit and support in the new api: https://github.com/pytorch/pytorch/pull/75135

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Review by andrewor14 — APPROVED** [review]

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

File: test/quantization/fx/test_quantize_fx.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/fx/test_quantize_fx.py:6012]: yeah I'm talking about new additions, not this one, please feel free to land this for urgent use cases

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35324817).
